src/result/result.py

```diff
@@ -441,3 +441,31 @@ async def async_wrapper(*args: P.args, **kwargs: P.kwargs) -> Result[R, TBE]:
         return async_wrapper
 
     return decorator
+
+
+def is_ok(result: Result[T, E]) -> TypeGuard[Ok[T]]:
+    """A typeguard to check if a result is an Ok
+
+    Usage:
+    >>> result = Ok(1)
+    >>> err = Err("error")
+    >>> if is_ok(result):
+    >>>     reveal_type(result)  # Revealed type is 'Ok[builtins.int*]'
+    >>> elif is_err(err):
+    >>>     reveal_type(err)  # Revealed type is 'Err[builtins.str*]'
+    """
+    return result.is_ok()
+
+
+def is_err(result: Result[T, E]) -> TypeGuard[Err[E]]:
+    """A typeguard to check if a result is an Err
+
+    Usage:
+    >>> result = Ok(1)
+    >>> err = Err("error")
+    >>> if is_ok(result):
+    >>>     reveal_type(result)  # Revealed type is 'Ok[builtins.int*]'
```

Comment:
As I understand it, `reveal_type` is a mypy feature, not a valid python function. Perhaps we can just add a comment suggesting the type of `result` is `Ok[int]`?
```suggestion
 >>>     result  # type of result is Ok[int]
```

Reply:
applied your version